payments-core/src/test/java/com/stripe/android/networking/StripeApiRepositoryTest.kt

```diff
@@ -2238,6 +2239,53 @@ internal class StripeApiRepositoryTest {
         assertThat(exception).isInstanceOf(InvalidRequestException::class.java)
     }
 
+    @Test
+    fun `Verify that the payment method options are persisted for dashboard card payments`() = runTest {
+        // Dashboard payments create a payment first
+        whenever(stripeNetworkClient.executeRequest(any<ApiRequest>()))
+            .thenReturn(
+                StripeResponse(
+                    200,
+                    PaymentMethodFixtures.CARD_JSON.toString(),
+                    emptyMap()
+                )
+            )
+
+        val confirmPaymentIntentParams =
+            ConfirmPaymentIntentParams.createWithPaymentMethodCreateParams(
+                paymentMethodCreateParams = PaymentMethodCreateParamsFixtures.DEFAULT_CARD,
+                clientSecret = "pi_12345_secret_fake",
+                paymentMethodOptions = PaymentMethodOptionsParams.Card(
+                    setupFutureUsage = ConfirmPaymentIntentParams.SetupFutureUsage.OffSession,
+                    moto = true
+                )
+            )
+
+        // Dashboard uses user key
+        create().confirmPaymentIntent(
+            confirmPaymentIntentParams = confirmPaymentIntentParams,
+            options = DEFAULT_OPTIONS.copy(
+                apiKey = "uk_12345"
+            )
+        )
+
+        verify(stripeNetworkClient, times(2))
```

Comment:
Why twice?

Reply:
Once for creating the payment method, and once for confirming.